Plutarch/Bool.hs

```diff
@@ -62,34 +62,26 @@ pnot = phoistAcyclic $ plam $ \x -> pif x (pcon PFalse) $ pcon PTrue
 infixr 3 #&&
 
 (#&&) :: Term s PBool -> Term s PBool -> Term s PBool
-x #&& y = pand # pdelay x # pdelay y
+x #&& y = pforce $ pand # x # pdelay y
 
 -- | Lazily evaluated boolean or for 'PBool' terms.
 infixr 2 #||
 
 (#||) :: Term s PBool -> Term s PBool -> Term s PBool
-x #|| y = por # pdelay x # pdelay y
+x #|| y = pforce $ por # x # pdelay y
 
 -- | Hoisted, Plutarch level, lazily evaluated boolean and function.
-pand :: Term s (PDelayed PBool :--> PDelayed PBool :--> PBool)
-pand = phoistAcyclic $
-  plam $
-    \x y -> pif' # pforce x # (pif' # pforce y # pcon PTrue # pcon PFalse) # pcon PFalse
+pand :: Term s (PBool :--> PDelayed PBool :--> PDelayed PBool)
+pand = phoistAcyclic $ plam $ \x y -> pif' # x # y # (phoistAcyclic $ pdelay $ pcon PFalse)
```

Comment:
Why do we need to hoist `pdelay $ pcon PFalse`  here (as well as in `por` below)?

Reply:
We don't need to. It will just reduce the size of the term.

Reply:
Maybe, but it doesn't seem obvious to me that variable reference is necessarily smaller than inlined `(\ -> False)`. I suppose we can measure these precisely once some benchmarks are in place (cf. https://github.com/Plutonomicon/plutarch/issues/79#issuecomment-1007473155 )

Reply:
It is indeed not obvious, and I am not 100% sure that this is the case either, but a variable reference ought to be smaller than `delay (constant ...)`, since the variable reference need not be any more than ~8 bits.

Reply:
In any case it's inlined if there's only one use of it.